### PR TITLE
feat(assert): add "and" function

### PR DESCRIPTION
**Pull request by zoncoen**
Add the "and" operator for assertions.

### CODE REVIEW

File: assert/operator_test.go [hunk 1]
```diff
@@ -6,6 +6,45 @@ import (
...
+				Equal(q, "one"),
+				NotZero(q),
+			},
+			ok: "one",
+			ng: "1",
+		},
+	}
+	for _, test := range tests {
+		test := test
+		or := And(test.assertions...)(q)
```
Comment (goccy) [inline assert/operator_test.go:38]: nit:

Perhaps we need to use "and" instead of "or" ?

**Review by goccy — APPROVED** [review]
LGTM !

File: assert/operator_test.go (hunk 1 above)
Comment (zoncoen) [reply assert/operator_test.go]: Thanks, I fixed it. PTAL 🙏

Comment (goccy) [reply assert/operator_test.go]: 👍

**Pull request merged**